mono/metadata/class.c

```diff
@@ -10526,8 +10533,10 @@ mono_generic_class_is_generic_type_definition (MonoGenericClass *gklass)
 void
 mono_class_setup_interface_id (MonoClass *klass)
 {
+	if (!MONO_CLASS_IS_INTERFACE (klass))
```

Comment:
Why do we assert at https://github.com/mono/mono/pull/6048/files#diff-f085caaf39e861140ff40b9ef0c516d9R3491 and not here?